prow/spyglass/lenses/links/links.go

```diff
@@ -91,23 +93,50 @@ func humanReadableName(name string) string {
 	return strings.Join(words, " ")
 }
 
-// Body renders link to logs.
-func (lens Lens) Body(artifacts []api.Artifact, resourceDir string, data string, config json.RawMessage, spyglassConfig config.Spyglass) string {
-	type link struct {
-		Name string
-		URL  string
+func clickableLink(link string, spyglassConfig config.Spyglass) string {
+	if strings.HasPrefix(link, "gs://") {
+		return spyglassConfig.GCSBrowserPrefix + strings.TrimPrefix(link, "gs://")
+	}
+	if strings.HasPrefix(link, "http://") || strings.HasPrefix(link, "https://") {
+		// Use https://google.com/url?q=[url] redirect to avoid leaking referer.
+		url := url.URL{
+			Scheme: "https",
+			Host:   "google.com",
+			Path:   "/url",
+		}
+		q := url.Query()
+		q.Add("q", link)
+		url.RawQuery = q.Encode()
+		return url.String()
 	}
+	return ""
+}
 
+type link struct {
+	Name string
+	URL  string
+	Link string
+}
+
+func toLink(jobPath string, content []byte, spyglassConfig config.Spyglass) link {
+	url := strings.TrimSpace(string(content))
+	return link{
+		Name: humanReadableName(filepath.Base(jobPath)),
+		URL:  url,
+		Link: clickableLink(url, spyglassConfig),
+	}
+}
+
+// Body renders link to logs.
+func (lens Lens) Body(artifacts []api.Artifact, resourceDir string, data string, config json.RawMessage, spyglassConfig config.Spyglass) string {
 	var links []link
 	for _, artifact := range artifacts {
 		content, err := artifact.ReadAtMost(bytesLimit)
```

Comment:
It seems weird to me that we're reading the first bytesLimit of every artifact and then encoding it as URL without any other validation/extraction. Is that really the intended behavior?

Reply:
Yes, the artifact body is expected to be a single line url or gs:// path. Only some artifacts (the ones with 'link.txt' suffix) will reach this area.

Example test using this feature: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/1401947321612111872

The "debugging links" section shows an example "master-and-node-logs.link.txt" file.